hwmgr-plugins/controller/hardwareplugin_controller.go

```diff
@@ -99,9 +104,111 @@ func (r *HardwarePluginReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+		cert, key, err := sharedutils.GetKeyPairFromSecret(ctx, r.Client, secretName, hwplugin.Namespace)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get certificate and key from secret: %w", err)
+		}
+
+		config.TLSConfig.ClientCert = sharedutils.NewStaticKeyPairLoader(cert, key)
+	}
+
+	if hwplugin.Spec.AuthClientConfig.OAuthClientConfig != nil {
+		oauthConf := hwplugin.Spec.AuthClientConfig.OAuthClientConfig
+		secret, err := sharedutils.GetSecret(ctx, r.Client, oauthConf.ClientSecretName, hwplugin.Namespace)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get oauth secret '%s': %w", oauthConf.ClientSecretName, err)
+		}
+
+		clientID, err := sharedutils.GetSecretField(secret, sharedutils.OAuthClientIDField)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get '%s' from oauth secret: %s", sharedutils.OAuthClientIDField, err.Error())
+		}
+
+		clientSecret, err := sharedutils.GetSecretField(secret, sharedutils.OAuthClientSecretField)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get '%s' from oauth secret: %s", sharedutils.OAuthClientSecretField, err.Error())
+		}
+
+		config.OAuthConfig = &sharedutils.OAuthConfig{
+			ClientID:     clientID,
+			ClientSecret: clientSecret,
+			TokenURL:     strings.TrimSuffix(oauthConf.URL, "/") + "/" + strings.TrimPrefix(oauthConf.TokenEndpoint, "/"),
+			Scopes:       oauthConf.Scopes,
+		}
+	}
+
+	// TODO: process hwplugin.Spec.AuthClientConfig.BasicAuthSecret when `Basic` authType is supported
+
+	return &config, nil
+}
+
 // validateHardwarePlugin verifies secure connectivity to the HardwarePlugin's apiRoot using mTLS.
 func (r *HardwarePluginReconciler) validateHardwarePlugin(ctx context.Context, hwplugin *pluginv1alpha1.HardwarePlugin) (bool, error) {
-	// TODO - complete
+
+	if hwplugin.Spec.AuthClientConfig == nil {
+		return false, fmt.Errorf("missing authClientConfig configuration")
+	}
+
+	// Construct OAuth client configuration
+	config, err := r.setupOAuthClientConfig(ctx, hwplugin)
+	if err != nil {
+		return false, fmt.Errorf("failed to setup OAuthClientConfig: %w", err)
+	}
+
+	// build OAuth client information if type is not ServiceAccount
+	cf := notifier.NewClientFactory(config, sharedutils.DefaultBackendTokenFile)
+	httpClient, err := cf.NewClient(ctx, hwplugin.Spec.AuthClientConfig.Type)
+	if err != nil {
+		return false, fmt.Errorf("failed to build client: %w", err)
+	}
+
+	// Validate apiRoot URL
+	apiRoot, err := url.Parse(hwplugin.Spec.ApiRoot)
+	if err != nil {
+		return false, fmt.Errorf("invalid apiRoot URL '%s': %w", hwplugin.Spec.ApiRoot, err)
+	}
+
+	// Build request to validation endpoint url
+	url := fmt.Sprintf("%s%s", apiRoot, sharedutils.HardwarePluginValidationEndpoint)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return false, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	// Send request to HardwarePlugin server
+	resp, err := httpClient.Do(req)
```

Comment:
This is ok for now, but what we should really be doing here is code-generating the client code for our API and using that to access the plugin API endpoints.  We'll have to do that anyway for when we actually access the plugin for 'real' stuff.

Reply:
100% agreed - I have this for now in order get the validation mechanism working in the absence of a HardwarePlugin server to test with. I have a PR ready that refactors the generated code and introduces the client code for the plugin API. I will add a "TODO comment here" to replace this segment with the generated version.